### PR TITLE
CORE-1257: convert all notification types to lower case

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: db/notification_types_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	tx, err := db.Begin()
+	assert.NoError(err, "unable to begin a transaction")
+	id, err := GetNotificationTypeID(tx, "test")
+	assert.NoError(err, "unexpected error occurred while looking up the notification type ID")
+	assert.Equal(testID, id)
+	tx.Rollback()
+
+	// Verify that all mock expectations were met.
+	assert.NoError(err, "not all mock expectations were met")
+}
```
Comment (slr71) [inline db/notification_types_test.go:36]: I was originally going to convert the notification type only when I was looking up the notification type ID, but upon closer inspection, that would have caused some problems. I added this test when I was updating the query, but I quickly decided that having `sqlmock` was eventually going to be very useful. I decided to adapt the test to work with the existing code and leave it.

File: handlers/legacy.go [hunk 2]
```diff
@@ -166,6 +166,7 @@ func (lh *Legacy) buildNotificationMessage(
 
 // HandleMessage handles a single AMQP delivery.
 func (lh *Legacy) HandleMessage(updateType string, delivery amqp.Delivery) error {
+	updateType = strings.ToLower(updateType)
```
Comment (slr71) [inline handlers/legacy.go:169]: This is the primary change. The update type will always be converted to lower case.

File: handlers/legacy_test.go [hunk 3]
```diff
@@ -315,3 +315,52 @@ func TestNotificationWithoutMessage(t *testing.T) {
...
+	}
+	assert.Equal("analysis", savedOutgoingMessage.Type, "incorrect notification type")
+
+	// Verify that the notification was published and check the notification type.
+	notification := messagingClient.PublishedNotificationMessage
+	if notification == nil {
+		t.Fatalf("no notification was published")
+	}
+	assert.Equal("analysis", notification.Message.Type, "incorrect notification type")
+}
```
Comment (slr71) [inline handlers/legacy_test.go:366]: This test verifies that the change has the desired effects.

**Review by johnworth — APPROVED** [review]
Looks good to me. Always entertaining that tests are far longer than the fix they're testing. :shipit:

**Comment by slr71** [comment]
Yep, it's so common for the unit testing code to be longer. 🤣 Thanks for the review!

**Pull request merged**